api/src/main/java/org/openmrs/api/db/ConceptDAO.java

```diff
@@ -120,6 +120,22 @@
 	        List<ConceptDatatype> excludeDatatypes, Concept answersToConcept, Integer start, Integer size)
 	        throws DAOException;
 	
+	/**
+	 * Return the number of concepts matching a search phrase and the specified arguments
+	 * 
+	 * @param phrase matched to the start of any word in any of the names of a concept
+	 * @param locales List&lt;Locale&gt; to restrict to
+	 * @param includeRetired Specifies whether to include retired concepts
+	 * @param requireClasses List&lt;ConceptClass&gt; to restrict to
+	 * @param excludeClasses List&lt;ConceptClass&gt; to leave out of results
+	 * @param requireDatatypes List&lt;ConceptDatatype&gt; to restrict to
+	 * @param excludeDatatypes List&lt;ConceptDatatype&gt; to leave out of results
+	 * @param answersToConcept all results will be a possible answer to this concept
+	 * @return the number of concepts matching the given search phrase
+	 * @throws APIException
+	 * @since 1.8
+	 * @should return a count of unique concepts
+	 */
 	public Integer getCountOfConcepts(String phrase, List<Locale> locales, boolean includeRetired,
```

Comment:
Too many parameters. Consider using a Search DTO instead. There are some services/DAOs using this pattern.

Reply:
@lluismf Would you mean the `@see` javadocs instead? :)

Reply:
No, I mean replacing the N parameters with a single one.

Reply:
Sorry, could you please elaborate? :D